torch_xla/csrc/helpers.cpp

```diff
@@ -383,10 +383,7 @@ xla::XlaOp XlaHelpers::PromotedBinaryOp(
     const xla::XlaOp& op1, const xla::XlaOp& op2,
     const std::function<xla::XlaOp(const xla::XlaOp&, const xla::XlaOp&)>&
         bin_op) {
-  xla::XlaOp numeric_op1 = ConvertToNumeric(op1);
-  xla::XlaOp numeric_op2 = ConvertToNumeric(op2);
-  std::pair<xla::XlaOp, xla::XlaOp> vops =
-      PromoteSecond(numeric_op1, numeric_op2);
+  std::pair<xla::XlaOp, xla::XlaOp> vops = PromoteSecond(op1, op2);
```

Comment:
This affect binary ops, should not be changed:
```
>>> a = torch.tensor(1, dtype=torch.int)
>>> b = torch.tensor(2, dtype=torch.long)
>>> c = a + b
>>> print(a)
tensor(1, dtype=torch.int32)
>>> print(b)
tensor(2)
>>> print(b.dtype)
torch.int64
>>> print(c)
tensor(3)
>>> print(c.dtype)
torch.int64
```

Reply:
@dlibenzi IIUC `ConvertToNumeric` only promote `PRED` to `U8` when applicable but does nothing else. 
https://github.com/pytorch/xla/blob/master/torch_xla/csrc/convert_ops.cpp#L56
I tried running the code block above with `torch/xla` master and it still returns =torch.int32`. So I think it's a separate issue which should be fixed in another PR?

Reply:
Oh right. This is OK. I thought you removed the `PromoteSecond()` call 😄

Reply:
Btw the type promotion logic in PyTorch will change along with the future merge of https://github.com/pytorch/xla/issues/820. I think we should fix the type promotion along with that breakage. The PT PR should be ready to merge soon I think :D